go.mod: bump Pebble to 1e8aa51077f0

jbowens opened the pull request:
```
1e8aa510 db: add Metric.Keys.MissizedTombstoneCount
4cbe9e79 sharedcache: EOF handling
d49a6965 batch: Add method to add internal keys
6f12cd7f docs: add machine-type benchmark annotation
c1c911a5 Skip TestManualCompaction test until fix is found (flaky)
a6e90217 *: add IngestAndExcise operation
898fb2f6 db: fix size adjustment for in-progress compactions
```

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104754)
<!-- Reviewable:end -->

jbowens committed:
go.mod: bump Pebble to 1e8aa51077f0

```
1e8aa510 db: add Metric.Keys.MissizedTombstoneCount
4cbe9e79 sharedcache: EOF handling
d49a6965 batch: Add method to add internal keys
6f12cd7f docs: add machine-type benchmark annotation
c1c911a5 Skip TestManualCompaction test until fix is found (flaky)
a6e90217 *: add IngestAndExcise operation
898fb2f6 db: fix size adjustment for in-progress compactions
```

Epic: none
Release note: none

jbowens commented:
tftr!

bors r+ single on
